MetaMorpheus/EngineLayer/FdrAnalysis/FdrAnalysisEngine.cs

```diff
@@ -254,20 +321,45 @@ private void QValueInverted(List<SpectralMatch> psms, bool peptideLevelAnalysis)
                 if (GlobalVariables.StopLoops) { break; }
-                int notch = psms[i].Notch ?? MassDiffAcceptorNumNotches;
-
-                // populate the highest q-Value for each notch 
-                if (!qValueNotchCalculated[notch])
-                {
-                    qValueNotch[notch] = (psms[0].GetFdrInfo(peptideLevelAnalysis).CumulativeDecoyNotch + 1) / psms[0].GetFdrInfo(peptideLevelAnalysis).CumulativeTargetNotch;
-                    qValueNotchCalculated[notch] = true;
-                }
 
                 qValue = Math.Min(qValue, (psms[i].GetFdrInfo(peptideLevelAnalysis).CumulativeDecoy + 1) / Math.Max(psms[i].GetFdrInfo(peptideLevelAnalysis).CumulativeTarget, 1));
-                qValueNotch[notch] = Math.Min(qValueNotch[notch], (psms[i].GetFdrInfo(peptideLevelAnalysis).CumulativeDecoyNotch + 1) / Math.Max(psms[i].GetFdrInfo(peptideLevelAnalysis).CumulativeTargetNotch, 1));
-
                 psms[i].GetFdrInfo(peptideLevelAnalysis).QValue = Math.Min(qValue, 1);
-                psms[i].GetFdrInfo(peptideLevelAnalysis).QValueNotch = Math.Min(qValueNotch[notch], 1);
+
+                // Notch ambiguous
+                if (psms[i].Notch is null)
+                {
+                    foreach (var notchGroup in psms[i].BestMatchingBioPolymersWithSetMods.GroupBy(p => p.Notch))
+                    {
+                        int notch = notchGroup.Key;
+                        var first = notchGroup.First();
+
+                        double cumDecoyNotch = peptideLevelAnalysis
+                            ? first.PeptideCumulativeDecoyNotch!.Value
+                            : first.CumulativeDecoyNotch!.Value;
+                        double cumTargetNotch = peptideLevelAnalysis
+                            ? first.PeptideCumulativeTargetNotch!.Value
+                            : first.CumulativeTargetNotch!.Value;
+
+                        qValueNotch[notch] = Math.Min(qValueNotch[notch], (cumDecoyNotch + 1) / Math.Max(cumTargetNotch, 1));
+                        foreach (var hyp in notchGroup)
+                        {
+                            if (peptideLevelAnalysis)
+                            {
+                                hyp.PeptideQValueNotch = Math.Min(qValueNotch[notch], 1);
+                            }
+                            else
+                            {
+                                hyp.QValueNotch = Math.Min(qValueNotch[notch], 1);
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    int notch = psms[i].Notch.Value;
+                    qValueNotch[notch] = Math.Min(qValueNotch[notch], (psms[i].GetFdrInfo(peptideLevelAnalysis).CumulativeDecoyNotch + 1) / Math.Max(psms[i].GetFdrInfo(peptideLevelAnalysis).CumulativeTargetNotch, 1));
+                    psms[i].GetFdrInfo(peptideLevelAnalysis).QValueNotch = Math.Min(qValueNotch[notch], 1);
+                }
```

Comment:
Does this duplicate lines 269-308? If so, should we break this out into a function?

Reply:
No. One is reversed from the other. Notice the Math.Min in the section you annotated versus the Max in the section you quoted. 

`int notch = psms[i].Notch.Value;
qValueNotch[notch] = Math.Max(qValueNotch[notch], psms[i].GetFdrInfo(peptideLevelAnalysis).CumulativeDecoyNotch / Math.Max(psms[i].GetFdrInfo(peptideLevelAnalysis).CumulativeTargetNotch, 1));
                    psms[i].GetFdrInfo(peptideLevelAnalysis).QValueNotch = Math.Min(qValueNotch[notch], 1);`